src/main/java/seedu/address/logic/parser/AddCommandParser.java

```diff
@@ -54,57 +54,113 @@ public AddCommand parse(String args) throws ParseException {
 
         switch (entityType) {
         case PERSON_ENTITY_STRING:
-            if (!arePrefixesPresent(
-                        argMultimap,
-                        PREFIX_NAME,
-                        PREFIX_ADDRESS,
-                        PREFIX_PHONE,
-                        PREFIX_STATUS,
-                        PREFIX_EMAIL)) {
-                throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddPersonCommand.MESSAGE_USAGE));
-            }
-
-            argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL,
-                    PREFIX_ADDRESS, PREFIX_STATUS);
-            Name name = ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get());
-            Phone phone = ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get());
-            Email email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
-            Address address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
-            Status status = ParserUtil.parseStatus(argMultimap.getValue(PREFIX_STATUS).get());
-            Set<Tag> tagList = ParserUtil.parseTags(argMultimap.getAllValues(PREFIX_TAG));
-
-            PersonDescriptor person = new PersonDescriptor(name, phone, email, address, status, tagList);
-
+            PersonDescriptor person = parseAddPerson(argMultimap);
```

Comment:
Excellent - this is what I meant by the SLAP.
The cut down into shorter methods should also help for readability!